src/cswinrt/code_writers.h

```diff
@@ -2178,17 +2180,46 @@ ComWrappersSupport.RegisterObjectForInterface(this, ThisPtr);
             write_activation_factory_objref_definition(w, class_type);
             auto objrefname = w.write_temp("%", bind<write_objref_type_name>(class_type));
 
-            w.write(R"(
-public %() : this(%(global::ABI.WinRT.Interop.IActivationFactoryMethods.ActivateInstanceUnsafe(%)))
+            if (settings.netstandard_compat)
+            {
+                w.write(R"(
+public %() : this(new %(global::ABI.WinRT.Interop.IActivationFactoryMethods.ActivateInstanceUnsafe(%)))
+{
+ComWrappersSupport.RegisterObjectForInterface(this, ThisPtr);
+}
+)",
+                    class_type.TypeName(),
+                    default_interface_name,
+                    objrefname);
+            }
+            else
+            {
+                bool has_base_type = !std::holds_alternative<object_type>(get_type_semantics(class_type.Extends()));
+                auto default_type_semantics = get_type_semantics(get_default_interface(class_type));
+                auto default_interface_typedef = for_typedef(w, default_type_semantics, [&](auto&& iface) { return iface; });
+                auto is_manually_gen_default_interface = is_manually_generated_iface(default_interface_typedef);
+
+                w.write(R"(
+public %() %
 {
+_inner = global::ABI.WinRT.Interop.IActivationFactoryMethods.ActivateInstanceUnsafe(%, %.IID);
 ComWrappersSupport.RegisterObjectForInterface(this, ThisPtr);
+ComWrappersHelper.Init(_inner, false);
 %
 }
 )",
-                class_type.TypeName(),
-                settings.netstandard_compat ? "new " + default_interface_name : "",
-                objrefname,
-                settings.netstandard_compat ? "" : "ComWrappersHelper.Init(_inner, false);");
+                    class_type.TypeName(),
+                    has_base_type ? ":base(global::WinRT.DerivedComposed.Instance)" : "",
+                    objrefname,
+                    bind<write_type_name>(default_type_semantics, typedef_name_type::StaticAbiClass, true),
+                    bind([&](writer& w)
+                    {
+                        if (is_manually_gen_default_interface)
+                        {
+                            w.write("_defaultLazy = new Lazy<%>(() => (%)new SingleInterfaceOptimizedObject(typeof(%), _inner));", default_interface_name, default_interface_name, default_interface_name);
```

Comment:
Not blocking on this PR, but I think we really should optimize these paths later. We're generating a bunch of lambdas, captures, and some overhead for `Lazy<T>`, etc. We can probably make this simpler and leaner.

Reply:
There are only 2 interfaces using this scenario now, I will probably be looking at removing them if possible later when I get to it.  I already had a change I was working on for this, will need to get back to it at some point.

Reply:
Oh that's awesome, sounds good!